streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -794,13 +794,11 @@ public class StreamsConfig extends AbstractConfig {
     /** {@code upgrade.from} */
     @SuppressWarnings("WeakerAccess")
     public static final String UPGRADE_FROM_CONFIG = "upgrade.from";
-    private static final String UPGRADE_FROM_DOC = "Allows upgrading in a backward compatible way. " +
-        "This is needed when upgrading from [0.10.0, 1.1] to 2.0+, or when upgrading from [2.0, 2.3] to 2.4+. " +
-        "When upgrading from 3.3 to a newer version it is not required to specify this config. Default is `null`. " +
-        "Accepted values are \"" + UPGRADE_FROM_0100 + "\", \"" + UPGRADE_FROM_0101 + "\", \"" +
-        UPGRADE_FROM_0102 + "\", \"" + UPGRADE_FROM_0110 + "\", \"" + UPGRADE_FROM_10 + "\", \"" +
-        UPGRADE_FROM_11 + "\", \"" + UPGRADE_FROM_20 + "\", \"" + UPGRADE_FROM_21 + "\", \"" +
-        UPGRADE_FROM_22 + "\", \"" + UPGRADE_FROM_23 + "\", \"" + UPGRADE_FROM_24 + "\", \"" +
+    private static final String UPGRADE_FROM_DOC = "Allows live upgrading (and downgrading) in a backward compatible way. " +
+        "This is needed for all upgrades from [2.4, 3.4] to 3.5 or above. " +
```

Comment:
based on my understanding that this config is still needed when upgrading across the 3.3<->3.4 and 3.4<->3.5 boundaries

Reply:
after digging up some other PRs I think there is maybe also a 3.2<->3.3 boundary. possibly also a 3.1 boundary though I'm not sure in which direction. hard to untangle all this

I guess for now we can just point to the upgrade guide and then follow up with an `upgrade.from` section there?

Reply:
> I guess for now we can just point to the upgrade guide and then follow up with an upgrade.from section there?

Yes, it might be good to add a new section to the docs and let this docs-string in the code just point to it. It's complicated, and I doubt we can explain it correctly in this docs string itself.